google/cloud/security/scanner/audit/instance_network_interface_rules_engine.py

```diff
@@ -0,0 +1,222 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Rules engine for NetworkInterface."""
+from collections import namedtuple
+import itertools
+import re
+
+from google.cloud.security.common.util.regex_util import escape_and_globify
+from google.cloud.security.common.util import log_util
+from google.cloud.security.scanner.audit import base_rules_engine as bre
+from google.cloud.security.scanner.audit import errors as audit_errors
+
+LOGGER = log_util.get_logger(__name__)
+
+class InstanceNetworkInterfaceRulesEngine(bre.BaseRulesEngine):
+    """Rules engine for InstanceNetworkInterfaceRules."""
+
+    def __init__(self, rules_file_path, snapshot_timestamp=None):
+        """Initialize.
+
+        Args:
+            rules_file_path: file location of rules
+        """
+        super(InstanceNetworkInterfaceRulesEngine,
+              self).__init__(rules_file_path=rules_file_path)
+        self.rule_book = None
+
+    def build_rule_book(self, global_configs=None):
+        """Build InstanceNetworkInterfaceRuleBook from rules definition file."""
+        self.rule_book = InstanceNetworkInterfaceRuleBook(
+            self._load_rule_definitions())
+
+    # pylint: disable=arguments-differ
+    def find_policy_violations(self, instance_network_interface,
+                               force_rebuild=False):
+        """Determine whether the networks violates rules."""
+        violations = itertools.chain()
+        if self.rule_book is None or force_rebuild:
+            self.build_rule_book()
+        resource_rules = self.rule_book.get_resource_rules()
+
+        for rule in resource_rules:
+            violations = itertools.chain(violations,
+                                         rule.find_violations(
+                                             instance_network_interface))
+        return violations
+
+    def add_rules(self, rules):
+        """Add rules to the rule book."""
+        if self.rule_book is not None:
+            self.rule_book.add_rules(rules)
+
+class InstanceNetworkInterfaceRuleBook(bre.BaseRuleBook):
+    """The RuleBook for enforced networks resources."""
+
+    def __init__(self,
+                 rule_defs=None):
+        """Initialize.
+
+        Args:
+            rule_defs: The parsed dictionary of rules from the YAML
+                definition file.
+            snapshot_timestamp: The snapshot to lookup data.
+        """
+        super(InstanceNetworkInterfaceRuleBook, self).__init__()
+        self.resource_rules_map = {}
+        if not rule_defs:
+            self.rule_defs = {}
+        else:
+            self.rule_defs = rule_defs
+            self.add_rules(rule_defs)
+
+    def add_rules(self, rule_defs):
+        """Add rules to the rule book."""
+        print(rule_defs)
```

Comment:
nit: please remember to remove print statements or convert to LOGGER statements.